src/pytorch_lightning/tuner/auto_gpu_select.py

```diff
@@ -55,6 +55,9 @@ def pick_single_gpu(exclude_gpus: List[int]) -> int:
     for i in range(num_cuda_devices()):
         if i in exclude_gpus:
             continue
+        # Check if the device has memory used by another torch process
+        if torch.cuda.memory_usage(f"cuda:{i}") > 0:
+            continue
```

Comment:
This check is redundant if the GPU is in exclusive mode, as only one process is allowed to run on it. That's why our check for placing a tensor on the device is enough to test for exclusivity. As the docs for memory_usage say, the check there is based on some sampling of ~1sec which is not fully reliable but we could probably still add it here. 

In addition, since you have been using/wanting to use this feature, would you mind checking out #13079 and comment on it with your thoughts? Thanks!